geonode/upload/upload.py

```diff
@@ -797,6 +789,16 @@ def final_step(upload_session, user, charset="UTF-8", dataset_id=None):
                     sld_uploaded = False
                 _log(f'[sld_uploaded: {sld_uploaded}] sld_file: {sld_file}')
 
+                if saved_dataset:
```

Comment:
@afabiani I don't think that this is necessary. The checks about the existence of the dataset is done later, the same as for the resource update, why we should do it again?

Reply:
@mattiagiupponi you are right, it's sufficient to get rid of this return here https://github.com/GeoNode/geonode/blob/7643ce88dc1c03fc12f2da90897df9e4eb3efda5/geonode/upload/upload.py#L913-L914

I will revert the changes accordingly